incubator/java-microprofile/image/project/Dockerfile

```diff
@@ -1,23 +1,34 @@
+# Step 1: Build the user's application
 FROM adoptopenjdk/openjdk8-openj9
 
 RUN apt-get update && \
     apt-get install -y maven unzip wget xmlstarlet
 
-COPY . /project
+# Copy and build the dev.appsody:java-microprofile parent pom
+COPY ./pom.xml /project/pom.xml
+RUN cd /project && mvn -B install dependency:go-offline -DskipTests
 
+# Copy and run a simple version check
+COPY ./util /project/util
 RUN  /project/util/check_version build
 
-RUN cd /project && mvn -B install dependency:go-offline -DskipTests
+# Copy the user's application pom and resolve all dependencies
+COPY ./user-app/pom.xml /project/user-app/pom.xml
+RUN cd /project/user-app && mvn package -Dskip=true -DskipTests
```

Comment:
Similar to the dockerfile-stack, validate.sh should be being driven here, to ensure the stack only builds final images that are based on maven projects that still have the parent reference intact.